test_vm/tests/account_authenticate_message_test.rs

```diff
@@ -0,0 +1,53 @@
+use fil_actor_account::types::AuthenticateMessageParams;
+use fil_actor_account::Method::AuthenticateMessage;
+use fvm_ipld_blockstore::MemoryBlockstore;
+use fvm_ipld_encoding::RawBytes;
+use fvm_shared::bigint::Zero;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::error::ExitCode;
+use test_vm::util::{apply_code, apply_ok, create_accounts, generate_deal_proposal};
+use test_vm::{TEST_VM_INVALID, VM};
+
+// Using a deal proposal as a serialized message, we confirm that:
+// - calls to authenticate_message with valid signatures succeed
+// - calls to authenticate_message with invalid signatures fail
+#[test]
+fn account_authenticate_message() {
+    let store = MemoryBlockstore::new();
+    let v = VM::new_with_singletons(&store);
+    let addr = create_accounts(&v, 1, TokenAmount::from(10_000e18 as i128))[0];
+
+    let proposal =
+        generate_deal_proposal(addr, addr, TokenAmount::zero(), TokenAmount::zero(), 0, 0);
+    let proposal_ser =
+        RawBytes::serialize(proposal).expect("failed to marshal deal proposal").to_vec();
+
+    // With a good sig, message succeeds
+    let authenticate_message_params = AuthenticateMessageParams {
+        signature: proposal_ser.clone(),
```

Comment:
I think something's amiss here - we haven't signed anything.

Reply:
The test harness treats anything that isn't `TEST_VM_INVALID` as a valid signature: https://github.com/filecoin-project/builtin-actors/blob/6df845dcdf9872beb6e871205eb34dcc8f7550b5/test_vm/src/lib.rs#L908-L920

I could make the signature be `vec![]` if that makes it more clear.

Reply:
Oh that is rather unexpected. Is there any scope to change the test harness to have a single _valid_ signature rather than invalid?

Reply:
Yeah, I'm happy to make that change. What I like to do is have the valid signature be the plaintext itself, and anything else is considered invalid. How do you feel about that?

Reply:
That works. It'll still be confusing at the call site to someone who hasn't followed through to read the comment explaining that, but it's basically ok once you know that.